lib/ansible/modules/source_control/gitlab_webhook.py

```diff
@@ -0,0 +1,442 @@
+        new_hook_attr = remote_webhook.attributes.copy()
+        new_hook_attr.update(local_webhook)
+        compare_on = ['url', 'events', 'token']
+        webhook_diff = not self.equals(remote_webhook.attributes, new_hook_attr, compare_on)
+        if webhook_diff:
+            self.setattrs(remote_webhook, local_webhook)
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+
+            remote_webhook.save()
+            return remote_webhook
+        else:
+            return False
+
+    def findByUrl(self, remote_webhooks, url):
+        found = []
+        for wh in remote_webhooks:
+            if wh.attributes['url'] == url:
+                found.append(wh)
+        return found
+
+    def asApiObject(self, url, events, hook_id=None, token=None):
+        obj = {
+            'url': url,
+            'push_events': bool('push' in events) and len(events) > 0,
+        }
+        for value in self.HOOK_EVENTS:
+            if value == 'push':
+                continue
+            obj[value + '_events'] = bool(value in events)
+        if hook_id:
+            obj['hook_id'] = hook_id
+        if token is not None:
+            obj['token'] = token
+        return obj
+
+    # ToDo: API does not provide current token of a given webhook
+    # https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/7220
+    def equals(self, prev_attr, new_attr, attributes):
+        for attr in attributes:
+            if attr == 'events':
+                for event in self.HOOK_EVENTS:
+                    if bool(prev_attr[event + '_events']) != bool(new_attr[event + '_events']):
+                        return False
+            elif attr == 'token':
+                if 'token' in new_attr and new_attr['token'] is not None:
+                    return False
+                else:
+                    continue
+            elif prev_attr[attr] != new_attr[attr]:
+                return False
+        return True
+
+    def setattrs(self, obj, new_attr):
+        for k, v in new_attr.items():
+            setattr(obj, k, v)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_url=dict(required=True),
+            validate_certs=dict(required=False, default=True, type='bool'),
+            api_username=dict(required=False, no_log=True),
+            api_password=dict(required=False, no_log=True),
+            api_token=dict(required=False, no_log=True),
+            hook_id=dict(required=False, type='int'),
+            solo=dict(required=False, default=True, type='bool'),
+            project=dict(required=True),
+            url=dict(required=False),
+            events=dict(required=False, type='list', elements='str', default=['push'],
+                        choices=['push', 'issues', 'confidential_issues', 'merge_requests', 'tag_push', 'note', 'job', 'pipeline', 'wiki_page']),
+            enable_ssl_verification=dict(required=False, default='yes', type='bool'),
```

Comment:
I would rather call this parameter `hook_ssl_verify` or something along those lines since it's related to the hook ssl verification and it doesn't seem very clear to me ( I know it's mentioned in the docs ). Also I don't seem to be able to find where this is used?